engine/src/main/java/io/camunda/zeebe/engine/state/message/DbMessageState.java

```diff
@@ -358,4 +368,16 @@ public boolean exist(
 
     return messageIdColumnFamily.exists(nameCorrelationMessageIdKey);
   }
+
+  private static boolean visit(
+      final long timestamp,
+      final ExpiredMessageVisitor visitor,
+      final DbCompositeKey<DbLong, DbForeignKey<DbLong>> compositeDeadlineKey) {
+    final long deadline = compositeDeadlineKey.first().getValue();
+    if (deadline <= timestamp) {
+      final long messageKey = compositeDeadlineKey.second().inner().getValue();
+      return visitor.visit(deadline, messageKey);
+    }
+    return false;
```

Comment:
:question: :x: Isn't that an issue now? If the deadline is larger then the current timestamp, this means we return false, which indicates visitor stopped, which indicates we should continue, which causes to immediately reschedule?

Reply:
You're right. This is a bug. Darn.

Thanks for catching ❤️

Reply:
I've cherry-picked the changes of #11959 onto this branch so the fix is immediately provided.

If that pull request is accepted, I'll consider this one accepted.